dna/src/commonTest/kotlin/predicate/PredicateFunctionContractTest.kt

```diff
@@ -0,0 +1,29 @@
+package com.github.trilobiitti.trilobite.dna.predicate
+
+import com.github.trilobiitti.trilobite.dna.data.DataTypes
+import com.github.trilobiitti.trilobite.dna.data.NullableType
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+
+class PredicateStrFunctionContractTest {
+
+    private val predicateStrFunctionContract = PredicateFunctionContractStrImpl();
+
+
+    @Test
+    fun shouldThrowsWhenPassNotValidPredicateFunctionInput(){
```

Comment:
```suggestion
    fun shouldThrowWhenInvalidPredicateFunctionInputIsPassed(){
```

also, this test doesn't cover all edge cases